packages/remix-server-runtime/server.ts

```diff
@@ -58,12 +56,45 @@ export const createRequestHandler: CreateRequestHandlerFunction = (
         );
       }
     });
+  return {
+    routes,
+    dataRoutes,
+    serverMode,
+    staticHandler,
+    errorHandler,
+  };
+}
+
+export const createRequestHandler: CreateRequestHandlerFunction = (
+  build,
+  mode
+) => {
+  let _build: ServerBuild;
+  let routes: ServerRoute[];
+  let serverMode: ServerMode;
+  let staticHandler: StaticHandler;
+  let errorHandler: HandleErrorFunction;
 
   return async function requestHandler(
     request,
     loadContext = {},
     { criticalCss } = {}
   ) {
+    _build = typeof build === "function" ? await build() : build;
```

Comment:
purposefully placing the `await` within `requestHandler` so that `createRequestHandler` doesn't need to change to `async` which would be breaking change.